generators/new_collateral/config/parser.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"errors"
+
+	"github.com/BurntSushi/toml"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/helpers"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/types"
+	"github.com/mitchellh/mapstructure"
+)
+
+type IParse interface {
+	ParseCurrentConfig(configFilePath, configFileName string) (types.TransformersConfig, error)
+}
+type Parser struct{}
+
+func (Parser) ParseCurrentConfig(configFilePath, configFileName string) (types.TransformersConfig, error) {
+	var tomlConfig types.TransformersConfigForToml
+	fullConfigFilePath := helpers.GetFullConfigFilePath(configFilePath, configFileName)
+	_, decodeErr := toml.DecodeFile(fullConfigFilePath, &tomlConfig)
+	if decodeErr != nil {
+		return types.TransformersConfig{}, decodeErr
+	}
+
+	metadata, metadataErr := parseExporterMetaData(tomlConfig)
+	if metadataErr != nil {
+		return types.TransformersConfig{}, metadataErr
+	}
+
+	transformerExporters, transformerExportersErr := parseTransformerExporters(tomlConfig)
+	if transformerExportersErr != nil {
+		return types.TransformersConfig{}, transformerExportersErr
+	}
+
+	return types.TransformersConfig{
+		ExporterMetadata:     metadata,
+		TransformerExporters: transformerExporters,
+		Contracts:            tomlConfig.Contracts,
+	}, nil
+}
+
+func parseExporterMetaData(tomlConfig types.TransformersConfigForToml) (types.ExporterMetaData, error) {
+	home, homeOk := tomlConfig.Exporter["home"].(string)
+	name, nameOk := tomlConfig.Exporter["name"].(string)
+	save, saveOk := tomlConfig.Exporter["save"].(bool)
+	schema, schemaOk := tomlConfig.Exporter["schema"].(string)
+	if !homeOk || !nameOk || !saveOk || !schemaOk {
+		return types.ExporterMetaData{}, errors.New("error asserting exporter meta data types")
+	}
+
+	var transformerNames []string
+	decodeErr := mapstructure.Decode(tomlConfig.Exporter["transformerNames"], &transformerNames)
+	if decodeErr != nil {
+		return types.ExporterMetaData{}, decodeErr
+	}
+
+	return types.ExporterMetaData{
+		Home:             home,
+		Name:             name,
+		Save:             save,
+		Schema:           schema,
+		TransformerNames: transformerNames,
+	}, nil
+}
+
+func parseTransformerExporters(tomlConfig types.TransformersConfigForToml) (types.TransformerExporters, error) {
+	delete(tomlConfig.Exporter, "home")
+	delete(tomlConfig.Exporter, "name")
+	delete(tomlConfig.Exporter, "save")
+	delete(tomlConfig.Exporter, "schema")
+	delete(tomlConfig.Exporter, "transformerNames")
```

Comment:
this makes me nervous 😅 

could you help me understand what we're doing here?

Reply:
Gonna take a whack to make sure I understand - this removes it from an in memory map right?

Reply:
TL;DR: Yep, like Eric said, this piece is removing these keys from an in-memory map. The reason I'm doing that is so it is possible to iterate through the rest of the key-value pairs in this map, which are `transformerExporters`, and then decode those into `TransformerExporter` structs. Since the `transformerExporters` have dynamic keys, it makes it a bit more challenging to auto-decode them as the config file is currently structured.

Longer explanation:
I think that this parser could be cleaned up quite a bit if we change the config file structure - currently we basically have `exporterMetadata` (e.g. "name", "save", "home", etc) and the `transformerExporters` (e.g. exporter.flip_eth_a_v1_1_0, etc) in the same `exporter` map in the config file. But since the `transformerExporter` keys are dynamic, I don't think that we can automatically decode the whole `exporter` map into a struct since the decoder won't know what field to put those `transformerExporters` in. The workaround I'm using in the parser, is to decode the whole `exporter` map into a `map[string]interface{}`, where the values are either the `exporterMetadata` fields, or the `transformerExporters` names.

Then `parseExporterMetaData` and `parseTransformerExporters` are the hacky methods to separate out that generic `exporter` map into two different maps, which can then be automatically decoded into `ExporterMetaData` and `TransformerExporters` structs, which make things ways easier to deal with while we're adding new transformers.

That ended up being longer than I meant it to, and would be happy to chat more about it outside of this PR! 😄